String exercises

danpottshimself opened the pull request:
I have completed the numbers challenges from the power point presentation at: https://tombola.atlassian.net/wiki/

Each task is separated by comments which explain what is being displayed.

dpotts7 committed:
Three of four exercises completed.

Added string exercises.

dpotts7 committed:
Completed string exercises.
I have added comments to display each task.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,32 @@
+//hello world
+console.log("hello world" + " JavaScript can use either “double quotes” or ‘single quotes’ to denote text.");
+//Extracting records from the following CSV
+var records = [1997, "Ford","E350",'"Super luxurious” truck'];
```
This is supposed to start as a string and end up as an array.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,32 @@
+console.log(word.search('[AEIOU]'));
+
+//Use a regex to determine if a given string is a number.
+var isANumber = isNaN("twenty")=== false;
```
question states regex?

dpotts7 committed:
Hello world exercise has been changed.

Returned as an array.

dpotts7 committed:
Made required changes.

Used ragex to determine if a string contains or is a number.

dpotts7 committed:
Changed the range value of the ragex match from 4 to 9.

This ensure it picks up all numbers.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,33 @@
+//Use a regex to determine if a given string is a number.
+var foo = function (quxx) {
+   quxx = "This contains 1 number";
+    var number = /[1-9]/g;
```
This won't work on the number 10, 100 or 0 for that matter ;)

dpotts7 committed:
Made changes to the range in the ragex to include 0.

Accidentally forgot to add 0 into the range.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,33 @@
+console.log(theArray[0]);
+
+//Extracting records from the following CSV
+var records = [1997, "Ford","E350",'"Super luxurious” truck'];
```
The idea is to start with a string and end up with an array.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,33 @@
+console.log(word.search('[AEIOU]'));
+
+//Use a regex to determine if a given string is a number.
+var foo = function (quxx) {
```
Naming - never leave foo bar etc in the code.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,33 @@
+//Use a regex to determine if a given string is a number.
+var foo = function (quxx) {
+   quxx = "This contains 1 number";
+    var number = /[0-9]/g;
```
Try this reges with the following:
19
1a

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,33 @@
+
+//Use a regex to determine Postcode validity
+var address = "I live at 19 lesbury avenue, NE28 7DR";
+var postcode_regex = /[A-Z]{1,2}[0-9][0-9A-Z]?\s?[0-9][A-Z]{2}/g;
```
The idea of this is to make sure the postcode is valid, not to see if the string contains a postcode.

dpotts7 committed:
I have made the changes as requested.
Three changes made.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,37 @@
+console.log(theArray);
+
+//Extracting records from the following CSV
+var records = [1997, "Ford","E350",'"Super luxurious” truck'];
```
Again, this is meant to start as a string and get parsed into an array.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,37 @@
+//Use a regex to determine if a given string is a number.
+var numberCheck = function (sentence) {
+   sentence = "This contains 1 number";
+    var number = /[0-9]/g;
```
try the following:
1
1.0
1a

dpotts7 committed:
Changed the cars into a string and then displayed it as an array in CSV format.

Changed the ragex term to use a + so that it looks out for other characters such as decimals.

keithbarrow commented on string-exercise.js:
```diff
@@ -0,0 +1,37 @@
+//Use a regex to determine if a given string is a number.
+var numberCheck = function (sentence) {
+   sentence = "This contains 1 number";
+    var number = /[0-9]+/g;
```
Still not right - try 
0a
1.2

keithbarrow commented:
Comments - just give me a shout when you've fixed the number thing.

dpotts7 committed:
fixed function:
checking if a string is a number rather than if a string contains a number.
used .test() functionality rather than .match

keithbarrow commented:
:shipit:
